File: benchmark/string.py
#!/usr/bin/python

import re
import time
import sys

str = 'abcdefgh'+'efghefgh'
imax = 1024 / len(str) * 512

starttime = time.time();
sys.stdout.flush()

gstr = ''
i = 0

while (i < imax+1000):
  i = i + 1
  gstr += str
  gstr = re.sub('efgh','____',gstr)
  lngth = len(gstr)
  if (lngth % (1024*64) == 0):
    print int(time.time()-starttime),"sec\t\t",(lngth/1024),"kb"
    sys.stdout.flush()

File: benchmark/string.pl
#!/usr/bin/perl -w

my $str='abcdefgh'.'efghefgh';
my $imax = 1024 / length($str) * 512;

my $starttime=time();

my $gstr='';
my $i=0;

while($i++ < $imax+1000)
{

        $gstr.=$str;
        $gstr=~s/efgh/____/g;
        my $lngth=length($gstr);   ##     my $lngth=length($gstr);        # Perhaps that would be a slower way
        print time()-$starttime,"sec\t\t",$lngth/1024,"kb\n" unless $lngth % (1024*64); #print out every 256kb
}
